akka-http-tests/src/test/scala/akka/http/scaladsl/marshalling/ContentNegotiationSpec.scala

```diff
@@ -19,147 +19,147 @@ import HttpCharsets._
 class ContentNegotiationSpec extends FreeSpec with Matchers {
   "Content Negotiation should work properly for requests with header(s)" - {
     "(without headers)" test { accept ⇒
-      accept(`text/plain` withCharset `UTF-16`) should select(`text/plain` withCharset `UTF-16`)
-      accept(`text/plain`, `text/html`) should select(`text/plain` withCharset `UTF-8`)
-      accept(`text/html`, `text/plain`) should select(`text/html` withCharset `UTF-8`)
+      accept(Seq(`text/plain` withCharset `UTF-16`)) should select(`text/plain` withCharset `UTF-16`)
```

Comment:
To avoid the extra verbosity, could we alternatively 
 * either, create a helper method right here on top which still takes a varargs param
 * or create a custom class instead of the plain Function1 which still has varargs params

Reply:
Implemented the second version which works without any syntax changes in 2.12+ but doesn't for 2.11 because SAM-support wasn't so good back then. Solved it with a small compat cross source file.